### DISCUSSION TITLE
How to make all markdown files be built into a GitBook site

### DISCUSSION THREAD

**Discussion by RheseSoemo** [GitbookIO/community#685] [0. General bugs and questions] [answered]
I'm using GitBook on a GitLab repo with a CI/CD pipeline. I was wanting all of my markdown files to be transformed into the site, even if not directly linked on the side panel. However, GitBook only seems to be building markdown into html if the page appears in the SUMMARY.md file and thus appear on the side panel.

Is there a setting or way to flag all markdown pages to be transformed into site pages, regardless on whether they appear in the SUMMARY.md file?

**Comment by addisonschultz** [comment]
GitBook should automatically index and generate summary.md based on the structure of your folder/repo - are there many markdown files that don't get uploaded?

**Reply by RheseSoemo** [reply]
No, all markdown files are in my remote repo on gitlab

**Comment by RheseSoemo** [comment]
No, they're all in the repository and I can navigate to the .md files manually on the site, but they aren't built into pages

**Reply by addisonschultz** [reply]
Are you able to share an example of the repo you're importing?

**Reply by addisonschultz** [reply]
It sound like the importer isn't working correctly in this case - seeing your repo can help us test/improve the importer in the future!

**Reply by RheseSoemo** [reply]
Yeah, I can make the repo public: https://gitlab.com/RheseSoemo/PortfolioV2Dev

**Comment by addisonschultz** [comment]
Thanks - and do you have a link to your published gitbook space?

**Reply by RheseSoemo** [reply]
Here's the link to the live site: https://rhesesoemo.gitlab.io/PortfolioV2Dev/
When I place items into the SUMMARY.md, I can navigate to them. EG: https://rhesesoemo.gitlab.io/PortfolioV2Dev/Personal/ComputerConstruction.html
However, when I don't place items in the SUMMARY.md the .md is still active on the site but no HTML is generated: EG: https://rhesesoemo.gitlab.io/PortfolioV2Dev/Corporate/WordpressProjects.html

**Comment by addisonschultz** [comment]
Ah - can you let me know the steps you're using to generate your portfolio? I believe you're using an old, now deprecated version of GitBook and the `gitbook` CLI - you should be using the web app now - which you can access at app.gitbook.com for free! 

Are you using the new version of GitBook?

**Reply by RheseSoemo** [reply]
I thought that might be the problem, so I used the gitbook platform to connect to the repository and pushed changes. However, I ran into similar issues, where I still couldn't access all of my pages.

**Reply by addisonschultz** [reply]
Gotcha - just so I can investigate this then: Can you share a few of the pages from https://gitlab.com/RheseSoemo/PortfolioV2Dev that do not show up in the sidebar? 

GitBook should do a pretty good job and automatically generate a listing in Summary.md with all of the md files it finds in a repository so I'd like to know which ones it's skipping

**Reply by RheseSoemo** [reply]
I've created a custom Summary.md to have items on the sidebar. What I'm wanting is some pages that I'm intentionally not showing on the sidebar to still be generated so that someone can navigate to them.

As an example, I have my page on [computer building](https://rhesesoemo.gitlab.io/PortfolioV2Dev/Personal/ComputerConstruction.html) in the side bar and it generates fine. However, I intentionally didn't list my page on [wordpress projects](https://rhesesoemo.gitlab.io/PortfolioV2Dev/Corporate/WordpressProjects.html) in the SUMMARY.md, but still want to be able to navigate to it.

**Reply by addisonschultz** [reply]
In order to have a page listed in GitBook, it needs to exist in Summary.md - but you can use hidden pages to potentially hide it from the sidebar, but you'll be able to navigate to it via direct link

https://docs.gitbook.com/content-editor/editor/content-structure/content-in-a-space#hide-a-page-from-the-table-of-contents

**Reply by RheseSoemo** [reply]
That's what I read on your guy's documentation, but when I try to do so, it doesn't work. 

I created the page [hidden test](https://gitlab.com/RheseSoemo/PortfolioV2Dev/-/blob/master/test/hidden-test.md) and marked it as hidden, but it still shows up on the navigation bar.

**Reply by addisonschultz** [reply]
Are you still referring to https://rhesesoemo.gitlab.io/PortfolioV2Dev/? Because the hidden page will only work with the latest version of GitBook

**Reply by RheseSoemo** [reply]
Yes, why wouldn't my repo have the latest version?

**Reply by addisonschultz** [reply]
https://rhesesoemo.gitlab.io/PortfolioV2Dev/ is using the old, now deprecated version of GitBook

**Reply by RheseSoemo** [reply]
How do I upgrade it to the new version then?

**Reply by addisonschultz** [reply]
First, you'll need to sign up for an account at ap.gitbook.com/join

**Reply by RheseSoemo** [reply]
I've already done that and messed around with it. Is that the only way to get the new version? I can't upgrade my CLI?

**Reply by addisonschultz** [reply, marked as answer]
The current version and the one you're using (deprecated/old version) are two separate applications. The CLI is no loner a part of the current GitBook tool.

In order to use the newest GitBook, please sign up for an account at app.gitbook.com/join. You'll be able to import your repository from GitLab via Git Sync. You can find more information on that here: https://www.youtube.com/watch?v=Ly2zQD0I9bY and https://docs.gitbook.com/integrations/git-sync

**Reply by RheseSoemo** [reply]
Alright, that makes sense. Thank you!
